### ISSUE TITLE
Update requirement of PHP to 7.2+

### ISSUE THREAD

**Issue by NamelessCoder** [TYPO3/Fluid#405]
This is a RFC to gather opinions about upgrading the PHP minimum version of Fluid to 7.2, in order to gain access to nullable types and nail down the entire API with strict return/argument types.

Currently Fluid supports PHP >= 5.5 so this is a rather dramatic upgrade. It would however in my opinion not make much sense to upgrade the requirement unless it is to 7.2 because of nullable types. Without those, there would still be (too many) places where a return type could still not be declared.

It might also be reasonable to compromise and *not* declare strict types in some areas of the API, e.g. ViewHelper methods which are guaranteed to be implemented in hundreds or even thousands of classes in production right now.

This change would obviously be breaking, thus the RFC. In particular I am interested in hearing if certain parts of the API (aside from VH methods as already mentioned) your particular Fluid-using framework would like to exempt.

And I am equally interested in hearing possible arguments for why PHP 7.0 for example would be sufficient.

Thanks in advance for any and all input!

**Comment by kitsunet** [comment]
No problem, please ping whenever you seem fit :)
We are currently at 7.1 and will probably update with next major in about a year. I think it's fine though, we can stay at the lastest < 7.2 version until then.
Generally I would like to see type hints. We started to add them in the framework as well. I currently don't know any particular place besides VH arguments that might be problematic. Everything else can probably be adapted.

**Comment by YetiCGN** [comment]
I've noticed you have set the minimum required PHP version in the [develop branch](https://github.com/TYPO3/Fluid/blob/6815505dde6042a5521de8bec228e5b91bf4baa5/composer.json#L7) already. Might I suggest you use a more constrained version range? Right now you're declaring compatibility with PHP 8.0, 9.0 and above - even though you probably haven't tested this package on those platforms. The version constraint `^7.2.0` is much more appropriate, since it at least sets the compatible version to `<= 8.0.0`. 

You can always increase the supported PHP version, but if you try to decrease it in a later version, people will simply not get any updates but be stuck on an old version of TYPO3/Fluid (or `typo3/fluid-engine`, then) which won't work with their PHP version either.

tl;dr: Please don't use version constraints without a sensible upper bound.

**Comment by NamelessCoder** [comment]
True and acknowledged, @YetiCGN!
